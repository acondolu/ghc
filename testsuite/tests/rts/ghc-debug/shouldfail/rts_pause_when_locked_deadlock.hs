{-# LANGUAGE ForeignFunctionInterface #-}

import Control.Concurrent
import Foreign
import Foreign.C
import System.Exit
import System.Timeout

foreign import ccall safe "deadlock.h assertDoneAfterOneSecond"
    safe_assertDoneAfterOneSecond_c :: Ptr CInt -> IO ()

foreign import ccall safe "deadlock.h pauseThenLock"
    safe_pauseThenLock_c :: Ptr CInt -> IO ()

main :: IO ()
main = alloca $ \donePtr -> do
  poke donePtr 0
  forkOS $ safe_assertDoneAfterOneSecond_c donePtr
  safe_pauseThenLock_c donePtr
