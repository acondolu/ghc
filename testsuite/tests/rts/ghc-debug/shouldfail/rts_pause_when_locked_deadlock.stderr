rts_pause_when_locked_deadlock: error: rts_lock: The RTS is already paused by this thread.
   There is no need to call rts_lock if you have already call rts_pause.
